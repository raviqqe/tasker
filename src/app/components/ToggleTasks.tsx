import { FaRegCheckSquare, FaRegSquare } from "react-icons/fa";
import { CircleButton } from "./CircleButton.js";
import { useLocation, useNavigate } from "react-router";

export const ToggleTasks = (): JSX.Element => {
  const { pathname } = useLocation();
  const navigate = useNavigate();
  const done = pathname === "/tasks/done";

  return (
    <CircleButton
      aria-label="Toggle Tasks"
      onClick={() => navigate(done ? "/tasks" : "/tasks/done")}
      secondary
    >
      {done ? <FaRegCheckSquare /> : <FaRegSquare />}
    </CircleButton>
  );
};
