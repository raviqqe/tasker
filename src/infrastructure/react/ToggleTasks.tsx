import { FaRegCheckSquare, FaRegSquare } from "react-icons/fa/index.js";
import { CircleButton } from "./CircleButton";

interface IProps {
  setTasksDone: (done: boolean) => void;
  tasksDone: boolean;
}

export const ToggleTasks = ({
  setTasksDone,
  tasksDone,
}: IProps): JSX.Element => (
  <CircleButton
    aria-label="Toggle Tasks"
    onClick={() => setTasksDone(!tasksDone)}
    secondary={true}
  >
    {tasksDone ? <FaRegCheckSquare /> : <FaRegSquare />}
  </CircleButton>
);
