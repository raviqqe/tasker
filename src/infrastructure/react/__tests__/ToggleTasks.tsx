import React from "react";
import { create } from "react-test-renderer";
import { ToggleTasks } from "../ToggleTasks";

it("renders", () => {
  expect(
    create(
      <ToggleTasks setTasksDone={() => {}} tasksDone={false} />
    ).toJSON()
  ).toMatchSnapshot();
});
