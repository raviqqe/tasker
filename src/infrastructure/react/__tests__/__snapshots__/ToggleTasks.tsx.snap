// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<button
  aria-label="Toggle Tasks"
  className="sc-bcXHqe sc-gswNZR hzLSTn gWKlTj"
  onClick={[Function]}
>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 448 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-6 400H54c-3.3 0-6-2.7-6-6V86c0-3.3 2.7-6 6-6h340c3.3 0 6 2.7 6 6v340c0 3.3-2.7 6-6 6z"
    />
  </svg>
</button>
`;
